content/en/blog/2021/wasm-progress/index.md

```diff
@@ -0,0 +1,68 @@
+In order to share the learning broadly, a [`wasm-extensions` repository](https://github.com/istio-ecosystem/wasm-extensions) was created under `istio-ecosystem` org. This repository serves two purposes:
+
+* Provides canonical example extensions, which also covers several highly demanded features such as [basic authentication](https://github.com/istio-ecosystem/wasm-extensions/tree/master/extensions/basic_auth).
+* Provides a guide for Wasm extension development, testing, and release. The guide is based on the same build tool chains and test frameworks that are used, maintained and tested by the Istio extensibility team.
+
+Currently the existing guide covers [WebAssembly extension development](https://github.com/istio-ecosystem/wasm-extensions/blob/master/doc/write-a-wasm-extension-with-cpp.md)
+and [unit testing](https://github.com/istio-ecosystem/wasm-extensions/blob/master/doc/write-cpp-unit-test.md) with C++,
+as well as [integration testing](https://github.com/istio-ecosystem/wasm-extensions/blob/master/doc/write-integration-test.md) with a GoLang test framework,
+which simulates real runtime by running a Wasm module with the Istio proxy binary.
+In the future, several more canonical extensions will also be added such as open policy agent, and header manipulation based on JWT token.
+
+## Wasm Module Distribution via Istio Agent
+
+Prior to Istio 1.9, [Envoy remote data source](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/base.proto#config-core-v3-remotedatasource) were needed to distribute remote Wasm modules to the proxy.
+An example configuration could be found [here](https://gist.github.com/bianpengyuan/8377898190e8052ffa36e88a16911910),
+where two `EnvoyFilter` resources are defined: one is to add a remote fetch Envoy cluster, and the other one is to inject a Wasm filter into HTTP filter chain.
+However this method has a drawback: if remote fetch fails either due to bad configuration or transient error, Envoy will be stuck with the bad configuration.
+If a Wasm extension is configured as [fail closed](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/wasm/v3/wasm.proto#extensions-wasm-v3-pluginconfig), a bad remote fetch will stop Envoy from serving.
+To fix this issue, [a fundamental change](https://github.com/envoyproxy/envoy/issues/9447) is needed to Envoy xDS protocol to make it allow asynchronous xDS response.
+
+Istio 1.9 provides a reliable distribution mechanism out of the box by leveraging the XDS proxy inside istio-agent and Envoy [Extension Configuration Discovery Server](https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/extension) (ECDS).
```

Comment:
just above we used `xDS`; here we are using `XDS`. should we just stick to `xDS` ?

Reply:
Thanks for catching this! Done.